framework/include/vectorpostprocessors/CylindricalAverage.h

```diff
@@ -0,0 +1,43 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef CYLINDRICALAVERAGE_H
+#define CYLINDRICALAVERAGE_H
+
+#include "SphericalAverage.h"
+
+class CylindricalAverage;
+
+template <>
+InputParameters validParams<CylindricalAverage>();
+
+/**
+ * Compute a cylindrical average of a variableas a function of radius throughout the
+ * simulation domain.
+ */
+class CylindricalAverage : public SphericalAverage
+{
+public:
+  CylindricalAverage(const InputParameters & parameters);
+
+protected:
+  /// compute the distance of the current quadarature point for binning
+  virtual Real computeDistance() override;
+
+  /// cylinder origin
+  const Point _origin;
```

Comment:
It might be worth moving that up into spherical average. We'd get the capability of shifted spheres for very little added work.